projects/adrv9009_zu11eg_som/carrier_bd.tcl

```diff
@@ -1,2 +1,88 @@
+ad_ip_parameter sys_audio_clkgen CONFIG.VCO_MUL 21
+ad_ip_parameter sys_audio_clkgen CONFIG.CLK0_DIV 85.5
+
+ad_connect sys_cpu_clk sys_audio_clkgen/clk
+ad_connect sys_i2s_mclk sys_audio_clkgen/clk_0
+
+# i2s ip
+ad_ip_instance axi_i2s_adi axi_i2s_adi
+ad_ip_parameter axi_i2s_adi CONFIG.DMA_TYPE 0
+ad_ip_parameter axi_i2s_adi CONFIG.S_AXI_ADDRESS_WIDTH 32
+
+# dma
+ad_ip_instance axi_dmac i2s_tx_dma
+ad_ip_parameter i2s_tx_dma CONFIG.DMA_TYPE_SRC 0
+ad_ip_parameter i2s_tx_dma CONFIG.DMA_TYPE_DEST 1
+ad_ip_parameter i2s_tx_dma CONFIG.CYCLIC 1
+ad_ip_parameter i2s_tx_dma CONFIG.AXI_SLICE_SRC 0
+ad_ip_parameter i2s_tx_dma CONFIG.AXI_SLICE_DEST 0
+ad_ip_parameter i2s_tx_dma CONFIG.ASYNC_CLK_DEST_REQ 0
+ad_ip_parameter i2s_tx_dma CONFIG.ASYNC_CLK_SRC_DEST 0
+ad_ip_parameter i2s_tx_dma CONFIG.ASYNC_CLK_REQ_SRC 0
+ad_ip_parameter i2s_tx_dma CONFIG.DMA_2D_TRANSFER 0
+ad_ip_parameter i2s_tx_dma CONFIG.DMA_DATA_WIDTH_DEST 32
+ad_ip_parameter i2s_tx_dma CONFIG.DMA_DATA_WIDTH_SRC 64
+
+ad_ip_instance axi_dmac i2s_rx_dma
+ad_ip_parameter i2s_rx_dma CONFIG.DMA_TYPE_SRC 1
+ad_ip_parameter i2s_rx_dma CONFIG.DMA_TYPE_DEST 0
+ad_ip_parameter i2s_rx_dma CONFIG.CYCLIC 1
+ad_ip_parameter i2s_rx_dma CONFIG.AXI_SLICE_SRC 0
+ad_ip_parameter i2s_rx_dma CONFIG.AXI_SLICE_DEST 0
+ad_ip_parameter i2s_rx_dma CONFIG.ASYNC_CLK_DEST_REQ 0
+ad_ip_parameter i2s_rx_dma CONFIG.ASYNC_CLK_SRC_DEST 0
+ad_ip_parameter i2s_rx_dma CONFIG.ASYNC_CLK_REQ_SRC 0
+ad_ip_parameter i2s_rx_dma CONFIG.DMA_2D_TRANSFER 0
+ad_ip_parameter i2s_rx_dma CONFIG.DMA_DATA_WIDTH_DEST 64
+ad_ip_parameter i2s_rx_dma CONFIG.DMA_DATA_WIDTH_SRC 32
+
+# i2s connections
+ad_connect sys_cpu_clk axi_i2s_adi/s_axi_aclk
+ad_connect sys_cpu_clk axi_i2s_adi/s_axis_aclk
+ad_connect sys_cpu_clk axi_i2s_adi/m_axis_aclk
+ad_connect sys_cpu_resetn axi_i2s_adi/s_axi_aresetn
+ad_connect sys_cpu_resetn axi_i2s_adi/s_axis_aresetn
+ad_connect i2s_tx_dma/m_axis axi_i2s_adi/s_axis
+
+#ad_connect i2s_rx_dma/s_axis axi_i2s_adi/m_axis
+# not connecting tlast
+ad_connect i2s_rx_dma/s_axis_data axi_i2s_adi/m_axis_tdata
+ad_connect i2s_rx_dma/s_axis_valid axi_i2s_adi/m_axis_tvalid
+ad_connect i2s_rx_dma/s_axis_ready axi_i2s_adi/m_axis_tready
+ad_connect i2s axi_i2s_adi/I2S
+ad_connect sys_i2s_mclk axi_i2s_adi/data_clk_i
+ad_connect sys_i2s_mclk i2s_mclk
+
+ad_connect sys_cpu_clk i2s_tx_dma/s_axi_aclk
+ad_connect sys_cpu_clk i2s_tx_dma/m_src_axi_aclk
+ad_connect sys_cpu_clk i2s_tx_dma/m_axis_aclk
+ad_connect sys_cpu_resetn i2s_tx_dma/s_axi_aresetn
+ad_connect sys_cpu_resetn i2s_tx_dma/m_src_axi_aresetn
+ad_cpu_interrupt ps-14 mb-14 i2s_tx_dma/irq
+
+ad_connect sys_cpu_clk i2s_rx_dma/s_axi_aclk
+ad_connect sys_cpu_clk i2s_rx_dma/m_dest_axi_aclk
+ad_connect sys_cpu_clk i2s_rx_dma/s_axis_aclk
+ad_connect sys_cpu_resetn i2s_rx_dma/s_axi_aresetn
+ad_connect sys_cpu_resetn i2s_rx_dma/m_dest_axi_aresetn
+ad_cpu_interrupt ps-15 mb-15 i2s_rx_dma/irq
+
+# interconnect
+ad_cpu_interconnect 0x80000000 i2s_rx_dma
```

Comment:
The addresses should be in the 0x4xxxxxxx range. i2s_rx_dma will overlap with the fan control, when both are merged.

Reply:
fixed addresses and interrupts